homeassistant/components/rainbird/switch.py

```diff
@@ -45,6 +56,22 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         devices.append(RainBirdSwitch(controller, switch, dev_id))
     add_entities(devices, True)
 
+    def start_irrigation(service):
+        """Set the duration device state attribute and start the irrigation."""
+        entity_id = service.data.get(ATTR_ENTITY_ID)
+        duration = service.data.get(ATTR_DURATION)
+
+        for device in devices:
+            if isinstance(device, RainBirdSwitch) and device.entity_id == entity_id:
+                device.start_irrigation(duration)
+
+    hass.services.async_register(
```

Comment:
We're in sync context so have to use the sync api. Use `hass.services.register`.

Reply:
yes